src/codecs/avif/encoder.rs

```diff
@@ -98,16 +98,24 @@ impl<W: Write> ImageEncoder for AvifEncoder<W> {
     /// The encoder currently requires all data to be RGBA8, it will be converted internally if
     /// necessary. When data is suitably aligned, i.e. u16 channels to two bytes, then the
     /// conversion may be more efficient.
+    #[track_caller]
     fn write_image(
         mut self,
         data: &[u8],
         width: u32,
         height: u32,
         color: ColorType,
     ) -> ImageResult<()> {
+        let expected_buffer_len =
+            (width as u64 * height as u64).saturating_mul(color.bytes_per_pixel() as u64);
         assert_eq!(
-            (width as u64 * height as u64).saturating_mul(color.bytes_per_pixel() as u64),
-            data.len() as u64
+            expected_buffer_len,
+            data.len() as u64,
+            "Invalid buffer length: expected {} got {} for image dimensions ({}, {})",
```

Comment:
Could you change the message to `"Invalid buffer length: expected {} got {} for {}x{} image"`? Beyond that, this looks good to me.

You can also write `"{width}"` and similar instead of passing a separate argument, though that only works for variables, not expressions like `data.len()`